src/adapter/src/coord/statement_logging.rs

```diff
@@ -254,10 +254,12 @@ impl Coordinator {
             (StatementLifecycleHistory, statement_lifecycle_updates),
             (SqlText, sql_text_updates),
         ] {
-            self.controller
-                .storage
-                .append_introspection_updates(type_, updates)
-                .await;
+            if !updates.is_empty() {
+                self.controller
+                    .storage
+                    .append_introspection_updates(type_, updates)
```

Comment:
Would it make sense to make `append_introspection_updates` just ignore any invocations with empty `updates`? Is there a valid reason why one would want empty updates to make it to the writer task? I think ticking forward the write frontier is not one because, as you say, that's already done periodically.

Reply:
I thought about that as well, but then decided against it because I din't want `append_introspection_updates` to silently ignore calls, even when they're empty, just in case they had semantic meaning. The way it is now, it's more explicit what's happening.

But yes, you're right about the second part, we don't need periodic ticking because the `CollectionManager` is doing that.